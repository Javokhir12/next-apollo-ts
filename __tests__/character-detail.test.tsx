import { GetCharacterByIdQuery } from '@/generated/graphql';
import renderer from 'react-test-renderer';
import Character from '@/pages/characters/[characterId]';

const characterProps: Pick<GetCharacterByIdQuery, 'character'> = {
  character: {
    gender: 'Male',
    id: 'abcdef777',
    image: 'https://test.com/rick.png',
    name: 'Rick',
    origin: {
      id: 'earth-777',
      name: 'The Earth',
    },
    status: 'Alive',
  },
};

describe('Character', () => {
  it('renders character info correctly', () => {
    const tree = renderer.create(<Character {...characterProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
