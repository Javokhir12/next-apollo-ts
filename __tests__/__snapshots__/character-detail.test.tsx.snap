// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character renders character info correctly 1`] = `
<article>
  <button
    className="chakra-button css-tevss8"
    type="button"
  >
    <a
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Back to Home
    </a>
  </button>
  <div
    className="css-1e9f2lr"
  >
    <span
      style={
        {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "inline-block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "width": "initial",
        }
      }
    >
      <span
        style={
          {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27350%27%20height=%27275%27/%3e"
          style={
            {
              "background": "none",
              "border": 0,
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        />
      </span>
      <img
        alt="Rick"
        data-nimg="intrinsic"
        decoding="async"
        onError={[Function]}
        onLoad={[Function]}
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          alt="Rick"
          data-nimg="intrinsic"
          decoding="async"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Ftest.com%2Frick.png&w=750&q=75"
          srcSet="/_next/image?url=https%3A%2F%2Ftest.com%2Frick.png&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Ftest.com%2Frick.png&w=750&q=75 2x"
          style={
            {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </span>
    <div
      className="css-bvqhxl"
    >
      <h2
        className="chakra-heading css-1imclce"
      >
        Rick
      </h2>
      <p
        className="chakra-text css-1n49pl7"
      >
        <b
          className="chakra-text css-0"
        >
          Origin:
        </b>
         
        The Earth
      </p>
      <p
        className="chakra-text css-1n49pl7"
      >
        <b
          className="chakra-text css-0"
        >
          Status:
        </b>
         
        Alive
      </p>
      <p
        className="chakra-text css-1n49pl7"
      >
        <b
          className="chakra-text css-0"
        >
          Gender:
        </b>
         
        Male
      </p>
    </div>
  </div>
</article>
`;
